enhancements/insights/config-api.md

```diff
@@ -0,0 +1,245 @@
+	Status string `json:"status,omitempty"`
+}
+
+```
+
+### User Stories
+
+- As an OCP user/cluster-admin I would like to be able to trigger on demand gathering (there were some active Insights health checks in the cluster, which should be fixed and I want to refresh the Insights status). 
+- As an OCP user/cluster-admin I would like to disable Insights gathering completely.
+- As an OCP user/cluster-admin I would like to see the latest Insights status.
+- As an OCP user/cluster-admin I would like to exclude/include particular gatherers.
+
+### API Extensions
+
+This is adding a new `insights.config.openshift.io` API. The proposal of the structure of the new API is mentioned in the [Proposal](#Proposal) section.
+
+### Risks and Mitigations
+
+Risk: Insights operator is not able to read the new config API.
+
+Mitigation: The Insights operator is marked as unvailable through the corresponding clusteroperator condition.
+
+Risk: Somebody/something is repeatedly updating the `ForceGatheringReason` attribute and thus forcing gathering. There is a risk that 
+the cluster would upload data too often, but this assumes that the "repeating interval" allows uploading (if it's forced too quickly then there will be no upload, because the gathering is not over).
+
+Mitigation: Maybe we should introduce some limit on the number of forced gatherings within 1 hour, for example. 
+
+## Design Details
+
+### On demand data gathering options
+
+The on demand data gathering can happen almost immediately when there is no active gathering in the progress, but when there is some in progress 
+then the question is what to do in this situation. There are following options:
+
+- schedule it - e.g using something like the `ScheduleGatheringAt` attribute where the user would provide some time in the future.
+  - Pros 
+    - no need to interrupt gathering in progress
+    - we could probably use the `RateLimiting` queue and add a new gathering to the queue with the `AddAfter` method. Otherwise we would need to define some waiting time (when there’s already some gathering in progress) and right now it’s difficult to estimate an average gathering time since the gatherers run in parallel
+  - Cons
+    - user has to define the scheduling time correctly in the future
+    - gathering doesn’t happen immediately
+- force it - e.g using a `ForceGatherinReason` attribute where the user would provide some string as reason (mentioned in the [Proposal](#proposal)). 
+  - pros
+    - [roviding a reason (e.g “Refresh Insights”) is simpler for configuration than providing a time
+    - gathering would be triggered almost immediately
+    - there is already similar precedent with `ForceRedeploymentReason` when redeploying static pods
+  - cons
+    - need to be able to interrupt gathering (if any in progress)
+
+The second option of forcing it is preferred and it also corresponds to the API proposal in the [Proposal](#proposal) section.  
```

Comment:
once you feel like you've waited long enough for feedback, probably go ahead and pick your favorite, work it into the proposal, and then move this section to an *Alternatives* subsection, where you lay out the pros and cons of the various options and explain why you went with the one you chose.

Reply:
Updated.